### ISSUE TITLE
Compilation Issue with Rust 1.58.1

### ISSUE THREAD

**Issue by Valkryst** [svenstaro/dummyhttp#248]
I'm encountering some build issues, which I'm unsure of how to resolve, when updating a GitHub Actions workflow from Rust 1.52.0 to 1.58.1. I believe these issues are caused by _dummyhttp_ as they occur when installing this crate.

Initially, the following error was thrown when installing _dummyhttp_ VIA `cargo install dummyhttp`:

```
Compiling dummyhttp v0.4.3
error: failed to load bc of "actix_codec-18f66ec5f257df0c.actix_codec.db48e966-cgu.0.rcgu.o": 

error: failed to compile `dummyhttp v0.4.3`, intermediate artifacts can be found at `/tmp/cargo-installYnF7Qj`

Caused by:
  could not compile `dummyhttp` due to previous error
Error: Process completed with exit code 101.
```

After a bit of unhelpful Google searching, I thought that the error might be caused by an outdated dependency of _dummyhttp_ and so I switched to installing from the `master` branch with `cargo install --git https://github.com/svenstaro/dummyhttp --branch master`.

This did change the error message (see below), but I'm still unable to determine the exact cause or how to resolve the issue.

```
warning: Linking COMDATs named '__profc__RNvMsR_NtCsbWg78tuRmzJ_5bytes5bytesNtB5_5Inner3len': nodeduplicate has been violated!

error: failed to load bc of "actix_codec-83e39a1e89119f5e.actix_codec.a0ab1b8f-cgu.0.rcgu.o": 

warning: `dummyhttp` (bin "dummyhttp") generated 1 warning
error: failed to compile `dummyhttp v0.4.3 ([https://github.com/svenstaro/dummyhttp?branch=master#5ac04a62)`,](https://github.com/svenstaro/dummyhttp?branch=master#5ac04a62)%60,) intermediate artifacts can be found at `/tmp/cargo-installaSyRJJ`

Caused by:
  could not compile `dummyhttp` due to previous error; 1 warning emitted
Error: Process completed with exit code 101.
```

Do you have any suggestions?

**Comment by svenstaro** [comment]
Can you try `cargo install --locked dummyhttp`?

**Comment by Valkryst** [comment]
Just gave that a shot

Here are the warnings, which occurred near the beginning of compilation:

```
warning: package `ahash v0.2.17` in Cargo.lock is yanked in registry `crates-io`, consider running without --locked
warning: package `arc-swap v0.4.3` in Cargo.lock is yanked in registry `crates-io`, consider running without --locked
warning: package `crossbeam-deque v0.7.2` in Cargo.lock is yanked in registry `crates-io`, consider running without --locked
warning: package `miow v0.2.1` in Cargo.lock is yanked in registry `crates-io`, consider running without --locked
warning: package `net2 v0.2.33` in Cargo.lock is yanked in registry `crates-io`, consider running without --locked
warning: package `quote v1.0.2` in Cargo.lock is yanked in registry `crates-io`, consider running without --locked
warning: package `socket2 v0.3.11` in Cargo.lock is yanked in registry `crates-io`, consider running without --locked
```

Here's the error message:

```
error: failed to load bc of "actix_codec-456e169c33736c12.actix_codec.10d7dd1f-cgu.0.rcgu.o": 

error: failed to compile `dummyhttp v0.4.3`, intermediate artifacts can be found at `/tmp/cargo-installCkZoy2`

Caused by:
  could not compile `dummyhttp` due to previous error
Error: Process completed with exit code 101.
```

**Comment by svenstaro** [comment]
Oh crap, well I suppose I'm going to update dummyhttp to actix 4 or something now that it's officially out.

**Comment by Valkryst** [comment]
I'll revert back to an older version of Rust until then. Thanks for responding so quickly!

**Comment by svenstaro** [comment]
This is now fixed with 0.5.0!
